### PR TITLE
Issue42

### PR DESCRIPTION
**Pull request by mssumanth**
Supported ranges for vlan in fvctl-json.py script

### CODE REVIEW

File: scripts/fvctl-json.py [hunk 1]
```diff
@@ -310,24 +312,54 @@ def do_addFlowSpace(gopts, opts, args):
...
-    for action in actions:
-        parts = action.split('=')
-        act = { 'slice-name' : parts[0], "permission" : parts[1] }
-        acts.append(act)
-    req['slice-action'] = acts
-    ret = connect(gopts, "add-flowspace", passwd, data=[req])  
-    if ret:
-        print "FlowSpace %s was added with request id %s." % (args[0], ret)
+    #Check for range in the options
+    if opts.rng == "vlan":
```
Comment (alshabib) [inline scripts/fvctl-json.py]: Why do I have to say -r vlan and not ust simply say dl_vlan=x-y. Specifying -r seems redundant.

File: scripts/fvctl-json.py [hunk 2]
```diff
@@ -717,6 +749,7 @@ def makeMatch(matchStr):
     for item in matchItems:
         it = item.split('=')
         if len(it) != 2:
+            print it
             print "Match items must be of the form <key>=<val>"
```
Comment (alshabib) [inline scripts/fvctl-json.py]: use this -> print "Match items must be of the form <key>=<val>, not %s" % it

File: scripts/logrotate [hunk 3]
```diff
@@ -1,4 +1,5 @@
-/var/log/flowvisor/flowvisor-stderr.log {
+/var/log/flowvisor/flowvisor-db.log /var/log/flowvisor/flowvisor-stderr.log {
+   weekly
```
Comment (alshabib) [inline scripts/logrotate:2]: did you add this? if yes why?

File: src/org/flowvisor/api/handlers/configuration/SetConfig.java [hunk 4]
```diff
@@ -12,9 +12,12 @@
 import org.flowvisor.config.SliceImpl;
 import org.flowvisor.config.SwitchImpl;
 import org.flowvisor.exceptions.MissingRequiredField;
+import org.flowvisor.exceptions.NoParam;
```
Comment (alshabib) [inline src/org/flowvisor/api/handlers/configuration/SetConfig.java]: Call this exception NoParamException

File: src/org/flowvisor/api/handlers/configuration/SetConfig.java [hunk 5]
```diff
@@ -67,13 +73,25 @@ public JSONRPC2Response process(Map<String, Object> params) {
...
 		
 	}
 
 
-    private void processFMLimit(Map<String, Object> fmlimit) 
+    private void checkForNoParams(Map<String, Object> params)
+    		throws NoParam{
+		if(params.isEmpty()){
+			FVLog.log(LogLevel.DEBUG, null, "No parameters are specified. Please input the parameters and the value you want them to be set");
+			throw new NoParam(" No parameters are specified. Please input the parameters and the value you want them to be set! ");
```
Comment (alshabib) [inline src/org/flowvisor/api/handlers/configuration/SetConfig.java]: Stop with the exclamation marks....

File: src/org/flowvisor/exceptions/SliceNameDisallowedException.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package org.flowvisor.exceptions;
+
+public class SliceNameDisallowedException extends Exception {
+	public SliceNameDisallowedException(String err) {
+		super(err);
+		// TODO Auto-generated constructor stub
```
Comment (alshabib) [inline src/org/flowvisor/exceptions/SliceNameDisallowedException.java]: Remove this todo

File: scripts/logrotate (hunk 3 above)
Comment (mssumanth) [inline scripts/logrotate:2]: Added this for log rotating derby db's log...

File: src/org/flowvisor/flows/FlowSpaceRuleStore.java [hunk 7]
```diff
@@ -98,7 +100,7 @@
 	 * This sorted structure maintains bitsets representing the rules for every
 	 * priority. Makes it easy to return the rule with the highest priority.
 	 */
-	private TreeMap<Integer, BitSet> prioSet = new TreeMap<Integer, BitSet>();
+	private static TreeMap<Integer, BitSet> prioSet = new TreeMap<Integer, BitSet>();
```
Comment (alshabib) [inline src/org/flowvisor/flows/FlowSpaceRuleStore.java]: why is this static?

**Pull request merged**